internal/querynodev2/segments/segment_loader.go

```diff
@@ -674,10 +667,14 @@ func (loader *segmentLoader) Load(ctx context.Context,
 		tr := timerecord.NewTimeRecorder("loadDurationPerSegment")
 		logger.Info("load segment...")
 
-		// L0 segment has no index or data to be load
+		// L0 segment has no index or data to be load.
 		if loadInfo.GetLevel() != datapb.SegmentLevel_L0 {
-			if err = loader.LoadSegment(ctx, segment.(*LocalSegment), loadInfo, loadStatus); err != nil {
-				return errors.Wrap(err, "At LoadSegment")
+			s := segment.(*LocalSegment)
+			// lazy load segment do not load segment at first time.
+			if !s.IsLazyLoad() {
```

Comment:
After remove `LoadStatus`.
All load path is controlled on here, but not distribute every load function with `LoadStatus` to avoid bugs.